### PR TITLE
vimc-4444: update commonly used db functions

### PR DESCRIPTION
**Pull request by xiangli313**
Youtrack:
https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-4444

Task:
This is to fix two functions.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+### vimpact 0.0.4
+https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-4444
+Fix extract_vaccine_history() and cohort_deaths_all_cause().
```
Comment (richfitz) [inline NEWS.md]:
```suggestion

* Fix extract_vaccine_history() and cohort_deaths_all_cause(). (VIMC-4444)
```

this is how we normally do news items if they reference issues. You can merge this suggestion directly from this comment (see buttons above)

File: R/commonly_used_db_data.R [hunk 2]
```diff
@@ -76,6 +77,10 @@ extract_vaccination_history <- function(con, touchstone_cov = "201710gavi", touc
   } else {
     p_int_pop <- external_population_estimates
   }
+
+  if(nrow(p_int_pop) == 0){
+    stop("No population data extracted - you might have specified invalid touchstone_pop.")
```
Comment (richfitz) [inline R/commonly_used_db_data.R:82]: it is useful to add tests to cover this sort of thing to make sure the error cases that you are trying to catch are caught. Now is really the best time to do it as you probably have a good reason for writing that.

**Review by richfitz — CHANGES_REQUESTED** [review]
it's hard to tell that this does the Right Thing with the current testing approach, but I get a sense of what is being done here.

Please update the news, and if it's easy to get tests covering the changes that would really help, but I understand this will be hard to do atm

File: R/commonly_used_db_data.R (hunk 2 above)
Comment (xiangli313) [reply R/commonly_used_db_data.R:82]: Thanks @richfitz. I will do as a seperate task. ATM, I am sure this is not an urgent issue comparing with the set of reports pending on vimpact.

**Review by richfitz — APPROVED** [review]

**Pull request merged**